icepyx/core/spatial.py

```diff
@@ -364,7 +410,22 @@ def validate_polygon_file(spatial_extent):
 
 
 class Spatial:
-    def __init__(self, spatial_extent, **kwarg):
+    _ext_type: ExtentType
+    _spatial_ext: list[float]
+    _geom_file: Optional[str]
```

Comment:
```suggestion
    _ext_type: ExtentType
    _geom_file: Optional[str]
    _spatial_ext: list[float]
```
I've been making them abc order - is there a best practice on this?

Reply:
0ca4485 reorders these attrs. 

In this case, however, I think we want `_spatial_ext` typed as only being a list of floats. Functions like `validate_polygon_list` take the user-supplied `spatial_extent` and convert it into a list of floats for consistency.